apps/property-manager/app/modules/properties/property/assets/blocks/delete/index.tsx

```diff
@@ -0,0 +1,93 @@
+import { useQueryClient } from '@tanstack/react-query'
+import type { Dispatch, SetStateAction } from 'react'
+import { toast } from 'sonner'
+import { useDeletePropertyBlock } from '~/api/blocks'
+import {
+	AlertDialog,
+	AlertDialogContent,
+	AlertDialogDescription,
+	AlertDialogHeader,
+	AlertDialogTitle,
+	AlertDialogFooter,
+	AlertDialogCancel,
+	AlertDialogAction,
+} from '~/components/ui/alert-dialog'
+import { Spinner } from '~/components/ui/spinner'
+import { QUERY_KEYS } from '~/lib/constants'
+
+interface Props {
+	data?: PropertyBlock
+	opened: boolean
+	setOpened: Dispatch<SetStateAction<boolean>>
+}
+
+export default function DeletePropertyBlockModal({
+	data,
+	opened,
+	setOpened,
+}: Props) {
+	const queryClient = useQueryClient()
+
+	const { mutate, isPending } = useDeletePropertyBlock()
+
+	const handleSubmit = () => {
+		if (data) {
+			mutate(
+				{
+					property_id: data?.property_id ?? '',
+					block_id: data?.id ?? '',
```

Comment:
you've already checked if data is available, i don't think you should do optional chaining and defaulting to ''??

```suggestion
				{
					property_id: data.property_id,
					block_id: data.id,
```